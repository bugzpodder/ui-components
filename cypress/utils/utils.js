/* eslint-disable */
import uuid from "uuid";

export const getRandomCharacters = length => {
  return uuid
    .v4()
    .replace(/-/g, "")
    .substring(0, length);
};

export const generateRandomNumber = () => {
  return Array(5)
    .fill(0)
    .map(() => Math.random() * 1000 * 1000 * 1000)
    .map(Math.round)
    .map(value => `${value}`)
    .map(value => value.padStart(9, "0"))
    .join("");
};

export const getTextFromElement = element => {
  if (typeof element === "string") {
    return cy.get(element).then(getTextFromElement);
  }
  return cy.wrap(
    Cypress.$(element)
      .text()
      .trim(),
  );
};

export const getActiveSpinner = () => {
  return cy.get("[data-testid='spinner-overlay'][data-is-active='true']");
};

export const waitForProgressIndicator = () => {
  return getActiveSpinner().should("not.exist");
};
