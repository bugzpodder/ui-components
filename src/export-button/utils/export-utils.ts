import {
  ExportableColumn,
  ReportOptions,
  toDelimitedReport,
} from "@grailbio/lib";
import { saveAs } from "file-saver";

export const generateReport = (
  fileName: string,
  columns: ExportableColumn<any>[],
  data: Record<string, any>[],
  options: ReportOptions = {},
): void => {
  const output = toDelimitedReport(columns, data, options);
  const { fileMimeType = "text/csv" } = options;
  const blob = new Blob([output], { type: `${fileMimeType};charset=utf-8;` });
  try {
    saveAs(blob, fileName, true);
  } catch (error) {
    console.error("error saving blob", error);
  }
};
