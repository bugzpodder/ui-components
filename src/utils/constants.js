// @flow
export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  value: suggestion.label,
  label: suggestion.label,
}));
