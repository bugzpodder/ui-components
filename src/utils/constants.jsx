// @flow
import React from "react";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const SERGEY =
  "https://media.licdn.com/dms/image/C5603AQGlBIPst1V4Ew/profile-displayphoto-shrink_800_800/0?e=1535587200&v=beta&t=2V2knyEhbTrhRWzgydsXLZqiX8gnoD1poj8jGeFpA0w";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];
