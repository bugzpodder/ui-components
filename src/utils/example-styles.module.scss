.container {
  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.spinnerContainer {
  position: relative;

  .buttonContainer {
    display: inline-block;
      position: relative;
  }

  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

