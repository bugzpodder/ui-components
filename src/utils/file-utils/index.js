// @flow
export { readFileListAsTextArray, readFileListAsText, readFileAsText } from "./file-utils";
