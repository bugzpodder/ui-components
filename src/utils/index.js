// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
