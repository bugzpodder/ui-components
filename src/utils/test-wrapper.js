// @flow
import React, { type Node } from "react";
// $FlowFixMe: there is no createGenerateClassName export in @material-ui/core/styles
import { createGenerateClassName } from "@material-ui/core/styles";
// $FlowFixMe: Cannot resolve module react-jss/lib/JssProvider
import JssProvider from "react-jss/lib/JssProvider";
import "@grail/lib/src/client-utils/mocks";

type Props = {
	children: Node<*>,
};

const generateClassName = createGenerateClassName({
	dangerouslyUseGlobalCSS: true,
});

export const TestWrapper = (props: Props) => {
	const { children } = props;
	return <JssProvider generateClassName={generateClassName}>{children}</JssProvider>;
};
