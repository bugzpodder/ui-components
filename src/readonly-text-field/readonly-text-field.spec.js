// @flow
import React from "react";
import { render } from "react-testing-library";
import { TestWrapper } from "../utils";
import { ReadOnlyTextField } from "./index";
import "jest-dom/extend-expect";

test("render readonly text field", async () => {
	const testText = "readOnly text";
	const { container, getByTestId } = render(
		<TestWrapper>
			<ReadOnlyTextField>{testText}</ReadOnlyTextField>
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent(testText);
	expect(container).toMatchSnapshot();
});

test("render readonly text field with empty children shown as -", async () => {
	const testText = "";
	const { container, getByTestId } = render(
		<TestWrapper>
			<ReadOnlyTextField>{testText}</ReadOnlyTextField>
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("-");
	expect(container).toMatchSnapshot();
});

test("render readonly text field with empty children", async () => {
	const testText = "";
	const { container, getByTestId } = render(
		<TestWrapper>
			<ReadOnlyTextField showEmptyValue>{testText}</ReadOnlyTextField>
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent("");
	expect(container).toMatchSnapshot();
});

test("render readonly text field with an icon", async () => {
	const testText = "Starred";
	const testIcon = "star";
	const { container, getByTestId } = render(
		<TestWrapper>
			<ReadOnlyTextField icon={testIcon}>{testText}</ReadOnlyTextField>
		</TestWrapper>,
	);
	expect(getByTestId("readonly-text-field")).toHaveTextContent(testText);
	expect(getByTestId("icon")).toHaveTextContent(testIcon);
	expect(container).toMatchSnapshot();
});
