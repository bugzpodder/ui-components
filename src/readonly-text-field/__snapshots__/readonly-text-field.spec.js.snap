// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render readonly text field 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    readOnly text
  </span>
</div>
`;

exports[`render readonly text field with an icon 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    <span
      aria-hidden="true"
      class="material-icons MuiIcon-root icon"
      data-testid="icon"
    >
      star
    </span>
    Starred
  </span>
</div>
`;

exports[`render readonly text field with empty children 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    
  </span>
</div>
`;

exports[`render readonly text field with empty children shown as - 1`] = `
<div>
  <span
    class="readonlyTextField"
    data-testid="readonly-text-field"
  >
    -
  </span>
</div>
`;
